sdgym/synthesizers/generate.py

```diff
@@ -125,6 +127,7 @@ def get_trained_synthesizer(self, data, metadata):
                 obj:
                     The trained synthesizer.
             """
+            metadata = SingleTableMetadata().load_from_dict(metadata)
```

Comment:
Bug the new test caught. The metadata is being passed as a dictionary, but  `get_trained_synthesizer_fn` expects the actual `Metadata` object.

I'll change these in another PR, where I'll pass the Metadata directly in the `benchmark_single_table` function, I just didn't want to do it here so it doesn't become too messy.

Reply:
It was decided this should be kept as is, since for these methods it is easier for the user to implement the metadata as a dictionary directly.